src/slot.rs

```diff
@@ -817,6 +822,9 @@ impl<Conn: Connection> SetWorker<Conn> {
     async fn run(&mut self) {
         loop {
             tokio::select! {
+                _ = &mut self.terminate_rx => {
+                    return;
+                },
```

Comment:
take it or leave it: giving the `SlotWorker` an explicit termination channel shouldn't _really_ be necessary if we instead sent a message over the request channel to tell it to shut down, I think?

not a big deal either way, a oneshot is pretty cheap.

Reply:
I agree with this in *spirit*, but I wanted the termination signal to short-circuit everything else. I didn't want a termination signal to get stuck in a queue of requests behind other (now irrelevant) work.

To this end -- I've started making some of these `select` loops biased, so that when we get a signal to exit, we don't try to do any other work.

Reply:
Yeah, that seems like a good call. Carry on!